### PR TITLE
Explicit save and back actions on reponse edit + go to next article on jaune edition

### PR DESCRIPTION
**Pull request by davidbgk**
The back button no longer needs the anchor/hash given that we don't want to highlight the line when we are coming back BUT keep the highlight when we save the response.

### CODE REVIEW

**Comment by ronnix** [comment]
We dont want the highlight in that case, but it would be nice to go back to our position in the list 🤔

File: repondeur/zam_repondeur/templates/amendements.html [hunk 1]
```diff
@@ -763,6 +770,7 @@ <h3>Supprimer</h3>
         editLink.addEventListener('click', e => {
             const thisURL = new URL(window.location.href)
             const linkURL = new URL(e.target.href)
+            thisURL.hash = ''
```
Comment (ronnix) [inline repondeur/zam_repondeur/templates/amendements.html]: The hash is not included in the value of `back` below, so I don't think this is needed

File: repondeur/zam_repondeur/models/article.py [hunk 2]
```diff
@@ -199,6 +199,22 @@ def create(
...
+        return sorted_articles[previous_index]
+
+    @property
+    def next_article(self) -> Optional["Article"]:
+        sorted_articles: List[Article] = sorted(self.lecture.articles)
+        next_index = sorted_articles.index(self) + 1
+        if next_index >= len(sorted_articles):
+            return None
+        return sorted_articles[next_index]
+
```
Comment (ronnix) [inline repondeur/zam_repondeur/models/article.py]: Maybe we should skip / filter out intersticial articles. We don't show them in the list as we don't need to edit info about them, so it's surprising to land on one when navigating to the next article.

Comment (davidbgk) [reply repondeur/zam_repondeur/models/article.py]: It is done if you want to take a look

File: repondeur/zam_repondeur/templates/amendements.html (hunk 1 above)
Comment (davidbgk) [reply repondeur/zam_repondeur/templates/amendements.html]: I think there is a tricky case when there is already a remaining hash coming from a previous back/save action which keep it. I'll double check

**Pull request merged**